src/main/java/servlets/admin/userManagement/DeletePlayers.java

```diff
@@ -0,0 +1,155 @@
+
+import dbProcs.Getter;
+import dbProcs.Setter;
+import utils.ShepherdLogManager;
+import utils.Validate;
+import utils.UserKicker;
+
+/**
+ * The delete players server side functionality is hanled by this control class
+ * <br/><br/>
+ * This file is part of the Security Shepherd Project.
+ * 
+ * The Security Shepherd project is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.<br/>
+ * 
+ * The Security Shepherd project is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.<br/>
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with the Security Shepherd project.  If not, see <http://www.gnu.org/licenses/>. 
+ * @author Cosmin Craciun
+ *
+ */
+public class DeletePlayers extends HttpServlet{
+	private static final long serialVersionUID = 1L;
+	private static org.apache.log4j.Logger log = Logger.getLogger(DeletePlayers.class);
+	
+	/** Initiated by deletePlayers.jsp.
+	 * Deletion of player from Core Database is handed by Setter.deleteUser
+	 * @param player Player ID to be deleted
+	 * @param csrfToken
+	 */
+	public void doPost (HttpServletRequest request, HttpServletResponse response) 
+						throws ServletException, IOException
+	{
+		//Setting IpAddress To Log and taking header for original IP if forwarded from proxy
+		ShepherdLogManager.setRequestIp(request.getRemoteAddr(), request.getHeader("X-Forwarded-For"));
+		log.debug("*** servlets.Admin.UserManagement.DeletePlayers ***");
+		PrintWriter out = response.getWriter();  
+		out.print(getServletInfo());
+		HttpSession ses = request.getSession(true);
+		Cookie tokenCookie = Validate.getToken(request.getCookies());
+		Object tokenParmeter = request.getParameter("csrfToken");
+		if(Validate.validateAdminSession(ses, tokenCookie, tokenParmeter))
+		{
+			ShepherdLogManager.setRequestIp(request.getRemoteAddr(), request.getHeader("X-Forwarded-For"), ses.getAttribute("userName").toString());
+			if(Validate.validateTokens(tokenCookie, tokenParmeter))
+			{
+				boolean notNull = false;
+				boolean validPlayer = false;
+				try
+				{
+					log.debug("Getting ApplicationRoot");
+					String ApplicationRoot = getServletContext().getRealPath("");
+					log.debug("Servlet root = " + ApplicationRoot );
+					
+					log.debug("Getting Parameters");
+					String player = request.getParameter("player");
+					log.debug("player = " + player);
+					
+					//Validation
+					notNull = (player != null);
+					if(notNull)
+					{
+						log.debug("Validating player " + player);
+						validPlayer = Getter.findPlayerById(ApplicationRoot, player);
+					}
+					if(notNull && validPlayer)
```

Comment:
The DeletePlayersTest tests that currently exist never actually gets past this line. There needs to be another test with an admin user that attempts to delete a different user tier account instead of themselves (Admins are currently not supported)

Reply:
Added new test method, testWithAdminAuthValidUser()